docs/update-database.rst

```diff
@@ -0,0 +1,98 @@
+Update a database
+=================
+
+A feature that is especially useful for databases growing over time,
+is the possibility to update a :class:`audformat.Database`.
+For instance, consider the following database that contains
+age labels for a hundred files.
+
+.. jupyter-execute::
+
+    import audformat
+    import audformat.testing
+
+
+    db = audformat.testing.create_db(minimal=True)
+    db.schemes['age'] = audformat.Scheme(
+        audformat.define.DataType.INTEGER,
+        minimum=20,
+        maximum=50,
+    )
+    audformat.testing.add_table(
+        db,
+        table_id='table',
+        index_type=audformat.define.IndexType.FILEWISE,
+        columns='age',
+        num_files=100,
+    )
+    db
+
+.. jupyter-execute::
+
+    db['table'].df
+
+Now assume we record some more files that we want add to our original database.
+To not confuse new and old data, we start from a fresh database containing
+only new files, get the annotations and update the original database.
+
+.. jupyter-execute::
+
+    db_update = audformat.testing.create_db(minimal=True)
+    db_update.schemes['age'] = db.schemes['age']
+    audformat.testing.add_table(
+        db_update,
+        table_id='table',
+        index_type=audformat.define.IndexType.FILEWISE,
+        columns='age',
+        num_files=range(101, 105),
+    )
+    db.update(db_update)  # update original database with new data
+    db['table'].df
+
+Or we find out that the some files have wrong labels.
+To update those, we again start from a fresh database containing only
+the critical files, relabel them and then update the original database.
```

Comment:
I would have loaded the original databases and used `.set(index=critical_files)` to update the labels, but I agree that it covers a meaningful example here.

Reply:
Yes, and if we think of the use-case where we relabel those files on iHUP, it might in fact be easier to just call `db.update()` instead of figuring out which table contains the altered label.